### PR TITLE
Avatar frames

### PR DESCRIPTION
**Pull request by hworld**

### CODE REVIEW

File: src/app/components/user/avatar-modal/UserAvatarModal.vue [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import FormUserAvatarFrame from '../../../../_common/user/user-avatar/frame/FormUserAvatarFrame.vue';
+import FormAvatar from '../../forms/avatar/avatar.vue';
+
+const modal = useModal()!;
+const { user } = useCommonStore();
+
+const hasAvatarFrameSelector = ref(false);
+
+onMounted(() => {
+	run(async () => {
```
Comment (hworld) [inline src/app/components/user/avatar-modal/UserAvatarModal.vue:18]: i think you can pass an async funciton into onMounted just fine without issues. no need for run()

File: src/app/components/user/AppUserAvatarBubble.vue [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	assetGuestImage,
+} from '../../../_common/user/user-avatar/AppUserAvatarImg.vue';
+import { User } from '../../../_common/user/user.model';
+import { styleBorderRadiusCircle, styleChangeBg } from '../../../_styles/mixins';
+import { ChatUser } from '../chat/user';
+import AppUserVerifiedWrapper from './AppUserVerifiedWrapper.vue';
+
+const props = defineProps({
+	user: {
+		type: [Object, null] as PropType<User | ChatUser | null>,
```
Comment (hworld) [inline src/app/components/user/AppUserAvatarBubble.vue:17]: null question again

File: src/app/components/user/AppUserAvatarBubble.vue [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	 *
+	 * Has no effect if {@link showFrame} is `false`.
+	 */
+	frameInset: {
+		type: Number,
+		validator: value => value === undefined || typeof value === 'number',
+		default: undefined,
+	},
+	/**
+	 * Smooshes it.
```
Comment (hworld) [inline src/app/components/user/AppUserAvatarBubble.vue:66]: better comment

Comment (hworld) [reply src/app/components/user/AppUserAvatarBubble.vue:66]: use chatgpt haha

File: src/app/components/user/AppUserAvatarBubble.vue [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+const href = computed(() => {
+	if (disableLink.value || !user.value) {
+		return undefined;
+	}
+
+	if (!link?.value) {
+		return Environment.wttfBaseUrl + user.value.url;
+	} else if (link.value === 'dashboard') {
+		return Environment.wttfBaseUrl;
+	} else if (link.value === 'fireside') {
```
Comment (hworld) [inline src/app/components/user/AppUserAvatarBubble.vue:101]: you don't need this anymore, we shouldn't support "fireside"

**Review by hworld — COMMENTED** [review]
change

File: src/_common/user/user-avatar/AppUserAvatarImg.vue [hunk 5]
```diff
@@ -1,8 +1,12 @@
-<script lang="ts" setup>
+<script lang="ts">
 import { PropType, ref } from 'vue';
 import { User } from '../user.model';
 import guestImage from './guest.png';
 
+export const assetGuestImage = guestImage;
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/_common/user/user-avatar/AppUserAvatarImg.vue:6]: made a new `images.ts` file in `_common/img/` and exported from there.

File: src/app/components/user/AppUserAvatarBubble.vue (hunk 4 above)
Comment (hapurubokka) [reply src/app/components/user/AppUserAvatarBubble.vue:101]: Should I remove this from `AppUserAvatarImg` as well then?

File: src/_common/user/user-avatar/frame/FormUserAvatarFrame.vue [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+	clearInterval(interval);
+});
+
+function pickFrame(frameId: number) {
+	if (!form.isLoaded) {
+		return;
+	}
+
+	form.formModel.avatar_frame = frameId;
+	form.changed = form.formModel.avatar_frame !== myFrameId.value;
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/_common/user/user-avatar/frame/FormUserAvatarFrame.vue]: Changed to use a form group and form control instead.

File: src/_common/message-thread/AppMessageThreadItem.vue [hunk 7]
```diff
@@ -1,36 +1,50 @@
...
+import { formatDate } from '../filters/date';
+import AppJolticon from '../jolticon/AppJolticon.vue';
+import { AppTimeAgo } from '../time/ago/ago';
+import AppTimelineListItem from '../timeline-list/item/item.vue';
+import AppUserCardHover from '../user/card/AppUserCardHover.vue';
+import { User } from '../user/user.model';
+
+defineProps({
+	user: {
+		type: [Object, null] as PropType<User | null>,
```
[the remark this answers is not in the document]
Comment (hapurubokka) [reply src/_common/message-thread/AppMessageThreadItem.vue:14]: allows `null` when used this way, not `any`

**Pull request merged**